src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +728,9 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+                Debug.Assert(!type.IsTupleType);
+                Debug.Assert(!other.IsTupleType);
```

Comment:
Added a comment and also reviewed the call paths.

You can get there either through `MatchSymbols.VisitNamedType` (which unwraps tuples into their underlying type) or `MatchSymbols.VisitMethod` (but that path only passes type arguments for comparison, see `MatchSymbols.AreMethodsEqual`).